policy.go

```diff
@@ -0,0 +1,67 @@
+package proxyproto
+
+import "net"
+
+// PolicyFunc can be used to decide whether to trust the PROXY info from
+// upstream. If set, the connecting address is passed in as an argument.
+//
+// See below for the different policies.
+//
+// In case an error is returned the connection is denied.
+type PolicyFunc func(upstream net.Addr) (Policy, error)
```

Comment:
I'm having no luck finding the issue that got reported to us with armon/go-proxyproto and a function in there similar to this, but I'm adding a comment here to say that I'm trying to find it because I believe the way for us to address it would have been to break the SourceChecker API in there, and I'm not sure whether the slightly different API here works around it. I'll keep digging and get back to you on that.

Reply:
To my knowledge - at least I didn't - ask for this. The author of the PR came up with the idea themselves. Feature wise, seems very similar what both of you are trying to achieve and I'm more than happy to include whatever you agree is the best option here.

Reply:
Yeah this is just a note to say that I might have more stuff to comment on here so please don't commit/merge this _quite_ yet. I just have to try to figure out where/what the report was. I'm striking out on both GitHub issues and customer issues, so far. Likely I'm just not getting keywords right.

Reply:
Sure, let's wait for the PR author to come back to you.

Reply:
See other comment; I think this is resolved.